zapisy/apps/defects/models.py

```diff
@@ -0,0 +1,60 @@
+from os.path import exists
+
+from django.conf import settings
+from django.core.files.storage import FileSystemStorage
+from django.db import models
+from django.contrib.auth.models import User
+from django.urls import reverse
+
+from gdstorage.storage import GoogleDriveStorage
+
+DEFECT_MAX_NAME_SIZE = 35
+DEFECT_MAX_PLACE_SIZE = 35
+
+
+class StateChoices(models.IntegerChoices):
+    CREATED = 0, "Zgłoszone"
+    WAITING = 1, "Oczekująca"
+    IN_PROGRESS = 2, "W realizacji"
+    DONE = 3, "Zakończone"
+
+
+class Defect(models.Model):
+    name = models.CharField(max_length=DEFECT_MAX_NAME_SIZE, verbose_name='Nazwa')
+    creation_date = models.DateTimeField(auto_now_add=True)
+    last_modification = models.DateTimeField(auto_now_add=True)
+    place = models.CharField(max_length=DEFECT_MAX_PLACE_SIZE, verbose_name="Miejsce")
+    description = models.TextField("Opis usterki", blank=True)
+    reporter = models.ForeignKey(User, on_delete=models.CASCADE)
+    state = models.PositiveSmallIntegerField("Stan", choices=StateChoices.choices, default=StateChoices.CREATED)
+    information_from_defect_manager = models.TextField("Informacja od serwisanta", blank=True)
+
+    def __str__(self):
+        return self.name
+
+    def get_url(self):
+        return reverse('defects:show_defect', args=[str(self.id)])
+
+    def get_status_color(self):
+        color = {StateChoices.CREATED: None, StateChoices.IN_PROGRESS: None, StateChoices.WAITING: None,
+                 StateChoices.DONE: "green"}[self.state]
+        return f"color: {color}" if color else ''
+
+
+def select_storage():
+    if exists(settings.GOOGLE_DRIVE_STORAGE_JSON_KEY_FILE):
+        return GoogleDriveStorage()
+    else:
+        FileSystemStorage(location="defect/")
```

Comment:
```suggestion
        return FileSystemStorage(location="defect/")
```
Dodatkowo, nawet jeśli to rozwiązanie to _fallback_, to w miarę możliwości powinno być jakoś działające. A zatem ścieżka powinna być po pierwsze absolutna, po drugie jakoś przemyślana – warto chyba wybrać coś w oparciu o `BASE_DIR` z `/zapisy/zapisy/settings.py`.

Reply:
Jedyny cel tego FileSystemStorage jest taki, żeby projekt się nie wywalał, gdy nie podamy pliku - docelowo tylko podczas pracy developerskiej w obszarach niedotykających tego pod serwisu. Z tego co pamiętam to pozostała część, nie jest do tego sposobu przystosowana, więc nie uważam naprawy tego(poza return) za opłacalną.

Reply:
Racja, to na pewno jest mniej ważne i niezbyt pilne, być może sam coś w tym pogrzebię w swoim czasie, jeśli będę uważał to za warte wysiłku (proszę więc nie _resolvować_ tego wątku). Natomiast coś, co musimy zrobić, jeśli faktycznie traktujemy ten `FileSystemStorage` jako _placeholder_, którego tak naprawdę nie zamierzamy używać, to nie wrzucać nic do niego, jeśli `select_storage` zwraca właśnie jego instancję. To wymaga prawdopodobnie przeciążenia metody `ImageForm.save`, ale też warunkowości w przygotowywaniu formularza tworzenia / edycji usterki.